drivers/ethernet/eth_mcux.c

```diff
@@ -394,19 +396,37 @@ static bool eth_get_ptp_data(struct net_if *iface, struct net_pkt *pkt,
 		if (ntohs(hdr_vlan->type) != NET_ETH_PTYPE_PTP) {
 			return false;
 		}
+
+		eth_hlen = sizeof(struct net_eth_vlan_hdr);
 	} else
 #endif
 	{
 		if (ntohs(NET_ETH_HDR(pkt)->type) != NET_ETH_PTYPE_PTP) {
 			return false;
 		}
+
+		eth_hlen = sizeof(struct net_eth_hdr);
 	}
 
 	net_pkt_set_priority(pkt, NET_PRIORITY_CA);
 
 	if (ptpTsData) {
 		/* Cannot use GPTP_HDR as net_pkt fields are not all filled */
-		struct gptp_hdr *hdr = gptp_get_hdr(pkt);
+		struct gptp_hdr *hdr;
+
+		/* In TX, the first net_buf contains the Ethernet header
```

Comment:
Well, I see that @tbursztyka approved this, but any comments? This looks too adhoc/fragile.